summarize/src/analysis.rs

```diff
@@ -164,3 +164,255 @@ pub fn perform_analysis(data: ProfilingData) -> Results {
         total_time,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use measureme::ProfilingDataBuilder;
+
+    #[test]
+    fn total_time_and_nesting() {
+        let mut b = ProfilingDataBuilder::new();
+
+        b.interval(QUERY_EVENT_KIND, "q1", 0, 100, 200, |b| {
+            b.interval(QUERY_EVENT_KIND, "q2", 0, 110, 190, |b| {
+                b.interval(QUERY_EVENT_KIND, "q3", 0, 120, 180, |_| {});
+            });
+        });
+
+        let results = perform_analysis(b.into_profiling_data());
+
+        assert_eq!(results.total_time, Duration::from_nanos(100));
+
+        // 10ns in the beginning and 10ns in the end
+        assert_eq!(results.query_data_by_label("q1").self_time, Duration::from_nanos(20));
+        // 10ns in the beginning and 10ns in the end, again
+        assert_eq!(results.query_data_by_label("q2").self_time, Duration::from_nanos(20));
+        // 60ns of uninterupted self-time
+        assert_eq!(results.query_data_by_label("q3").self_time, Duration::from_nanos(60));
+
+        assert_eq!(results.query_data_by_label("q1").invocation_count, 1);
+        assert_eq!(results.query_data_by_label("q2").invocation_count, 1);
+        assert_eq!(results.query_data_by_label("q3").invocation_count, 1);
+    }
+
+    #[test]
+    fn events_with_same_starting_time() {
+        //                      <--e4-->
+        //                      <---e3--->
+        //  <--------e1--------><--------e2-------->
```

Comment:
These are super helpful for understanding the tests. Thanks for adding them!

Reply:
Yes, they are super helpful for writing the tests too `:)`